### ISSUE TITLE
Annotation warnings (CS8632, CS8669) should not be reported with -nowarn:nullable

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#42344]
Warnings for the following should be suppressed with `-nowarn:nullable`:

```C#
class A
{
    static object? F1() => null;
}
```
```C#
// <auto-generated/>
class B
{
    static object? F2() => null;
}
```

`csc /t:library /nowarn:nullable a.cs b.cs`
```
a.cs(3,18): warning CS8632: The annotation for nullable reference types
    should only be used in code within a '#nullable' annotations context.
b.cs(4,18): warning CS8669: The annotation for nullable reference types
    should only be used in code within a '#nullable' annotations context. Auto-generated code
    requires an explicit '#nullable' directive in source.
```

**Comment by cston** [comment]
See https://github.com/dotnet/runtime/pull/33230

cc @safern, @jcouv

**Comment by jcouv** [comment]
@cston  does this repro with recent compiler?
I thought I fixed with https://github.com/dotnet/roslyn/pull/41686 already

**Comment by safern** [comment]
It repro with version `3.5.0-beta4-20153-05`.

**Comment by jcouv** [comment]
That makes sense. The fix went into 3.6 only (corresponding to VS 16.6), so 3.5 packages would exhibit the old behavior for NoWarn/WarnAsError:nullable.

**Comment by safern** [comment]
Should/Can we start using 3.6 in arcade then so that we don't hit this?

**Comment by safern** [comment]
Actually is the fix in master? currently arcade depends on: release/dev16.5-vs-deps -> .NET Core SDK 3.1.2xx

Which would be the correct branch to consume it from? This are the branches and the channels you guys publish to:
(546)  https://github.com/dotnet/roslyn @ release/dev16.3-vs-deps -> .NET Core 3 Release
(949)  https://github.com/dotnet/roslyn @ master-vs-deps -> .NET Core 5 Dev
(950)  https://github.com/dotnet/roslyn @ mattgal-roslyn -> General Testing
(991)  https://github.com/dotnet/roslyn @ TestArcadeSDK -> General Testing
(992)  https://github.com/dotnet/roslyn @ release/dev16.5-vs-deps -> .NET Core SDK 3.1.2xx
(879)  https://github.com/dotnet/roslyn @ release/dev16.4-vs-deps -> .NET Core SDK 3.1.1xx

cc: @jaredpar @agocke

**Comment by safern** [comment]
I can confirm that: `3.6.0-2.20160.3` doesn't have this issue.

**Comment by jcouv** [comment]
Cool, thanks for confirming. I'll go ahead and close the issue.

**Comment by safern** [comment]
Could someone confirm that moving to master branch is fine in the arcade subscription?

**Comment by jcouv** [comment]
> Could someone confirm that moving to master branch is fine in the arcade subscription?

Tagging @jaredpar @tmat @agocke for Santiago's question on arcade

**Comment by agocke** [comment]
I think that's fine. You'll need to grab from the NET Core 5 Dev branch for that.

**Comment by safern** [comment]
Ok, I'll do that then, thanks for confirming.

**Comment by safern** [comment]
Updated and triggered the subscription: https://github.com/dotnet/arcade/pull/5061
